src/main/java/org/cryptomator/ui/mainwindow/MainWindowTitleController.java

```diff
@@ -53,22 +54,45 @@
 
 	@FXML
 	public void initialize() {
-		LOG.debug("init MainWindowTitleController");
+		LOG.trace("init MainWindowTitleController");
 		updateChecker.automaticallyCheckForUpdatesIfEnabled();
 		titleBar.setOnMousePressed(event -> {
 			xOffset = event.getSceneX();
 			yOffset = event.getSceneY();
+
+		});
+		titleBar.setOnMouseClicked(event -> {
+			if (event.getButton().equals(MouseButton.PRIMARY)) {
+				if (event.getClickCount() == 2) {
+					if (window.isFullScreen()) {
+						window.setFullScreen(false);
+					} else {
+						window.setFullScreen(true);
+					}
+				}
+			}
 		});
 		titleBar.setOnMouseDragged(event -> {
+			if (window.isFullScreen()) return;
 			window.setX(event.getScreenX() - xOffset);
 			window.setY(event.getScreenY() - yOffset);
 		});
+		titleBar.setOnMouseReleased(event -> {
+			saveWindowSettings();
```

Comment:
Everytime someone clicks at the title bar, we are saving settings. Listen to the [`onMouseDragExitProperty`](https://openjfx.io/javadoc/16/javafx.graphics/javafx/scene/Node.html#onMouseDragExitedProperty()) instead.